swig/python2/examples/rpc_example.py

```diff
@@ -62,15 +62,14 @@ def print_value(value):
     else:
         print "(unprintable)"
 
-def test_rpc_cb(xpath, input_vals, input_cnt, output_vals, output_cnt, private_ctx):
+def test_rpc_cb(xpath, in_vals, out_vals, private_ctx):
 #    try:
         print "\n\n ========== RPC CALLED =========="
-        in_vals = sr.Vals(input_vals, input_cnt)
-        out_vals = sr.Vals(output_vals, output_cnt, 3)
-
         if (in_vals == None or out_vals == None):
             return
 
+        out_vals.allocate(3)
+
         for n in range(in_vals.val_cnt()):
             print_value(in_vals.val(n))
 
```

Comment:
_Regarding the counter in the C structs, what should I do for now?
Wait for the counter in the structs or implement the method with the "free" flag._

I didn't mean to make a usage counter inside the internal structs, but to add some between them and the wrapper class, i.e. on the C++ level. So you could use shared_ptr for the task, or, if you experience any issues, you could possibly create your own usage counter tailor made for the C++ to Python conversion using SWIG.